corehq/apps/locations/models.py

```diff
@@ -680,7 +680,11 @@ def save(self, *args, **kwargs):
         kwargs['sync_to_sql'] = False  # only sync here
         super(Location, self).save(*args, **kwargs)
         if sync_to_sql:
-            self._migration_do_sync()
+            try:
+                self._migration_do_sync()
+            except Exception:
+                self.delete()  # roll back on failure to sync
```

Comment:
Ah you're right.  I suppose what I really want is to revert to the previous rev if one exists, otherwise delete.  That sounds messy, know of any existing solutions to do that?